### ISSUE TITLE
`interfacer` doesn't preserve variadic arguments

### ISSUE THREAD

**Issue by jinnovation** [rjeczalik/interfaces#7]
Input struct: [`"github.com/gocql/gocql".Session`](https://godoc.org/github.com/gocql/gocql#Session)

Output:
``` go
// Session is an interface generated for "github.com/gocql/gocql".Session.
type Session interface {
// ...
	Query(string, []interface{}) *gocql.Query
// ...
}
```

Corresponding function signature (see [`github.com/gocql/gocql/session.go`](https://github.com/gocql/gocql/blob/master/session.go#L292):
``` go
func (s *Session) Query(stmt string, values ...interface{}) *Query
```

**Comment by rjeczalik** [comment]
Actually the `[]interface{}` is what what `go/types` package returns, not interfacer package. I acn take a look if there's any field that can be used to make a distinction.

For now workaround - edit the signature manually 😇 

Thanks @jinnovation for the report!

**Comment by sbward** [comment]
Hey there, this issue is really old, but there is a way to fix this using types.Signature.Variadic() method if you want to do the formatting manually.

**Comment by rjeczalik** [comment]
@sbward Thanks for pointing out, I must have missed that. Would you like to try sending a PR for that?

**Comment by sbward** [comment]
In my fork, I attempted to do it while preserving your structer/interfacer shared formatting code, but in the end I decided to just use gotypes Signature String formatter and delete structer.  As a result my fork is now just a simpler implementation of interfacer by itself.  It's up to you to decide if you want to take a similar approach and break apart interfacer/structer implementations, or bite the bullet and figure out how to use Signature.Variadic in the existing codebase.
